mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -132,11 +132,11 @@ def suggest(self, context: Optional[pd.DataFrame] = None, defaults: bool = False
             configuration = self._suggest(context)
             assert len(configuration) == 1, \
                 "Suggest must return a single configuration."
-            assert len(configuration.columns) == len(self.optimizer_parameter_space.values()), \
+            assert set(configuration.columns).issubset(set(self.optimizer_parameter_space)), \
```

Comment:
Is this too lenient now?

Reply:
I think it's good enough as a sanity check - in fact, it's better than checking just the number of hyperparameters. I don't know an easy way to make sure we have a consistent set of hyperparameters that satisfy all ConfigSpace conditionals while not omitting any required values. I'll do some research and maybe send another PR later